libs/git_wrapper/OGit_wrapper.ml

```diff
@@ -0,0 +1,159 @@
+(* Austin Theriault
+ *
+ * Copyright (C) Semgrep, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* Commentary *)
+(* This is like git_wrapper but using pure ocaml https://github.com/mirage/ocaml-git. *)
+(* That way we don't have to shell out *)
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+
+module Store = Git_unix.Store
+module Hash = Store.Hash
+module Value = Git.Value.Make (Hash)
+module Commit = Git.Commit.Make (Hash)
+module Tree = Git.Tree.Make (Hash)
+module Blob = Git.Blob.Make (Hash)
+module User = Git.User
+
+let tags = Logs_.create_tags [ __MODULE__ ]
+
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+type object_table = (Value.hash, Value.t) Hashtbl.t
+type hash = Hash.t [@@deriving show, eq, ord]
+type commit = Commit.t [@@deriving show, eq, ord]
+type blob = Blob.t [@@deriving show, eq, ord]
+type author = User.t [@@deriving show, eq, ord]
+
+type blob_with_extra = { blob : blob; path : Fpath.t; size : int }
+[@@deriving show]
+
+(*****************************************************************************)
+(* Reexports *)
+(*****************************************************************************)
+
+let commit_digest = Commit.digest
+let commit_author = Commit.author
+let hex_of_hash = Hash.to_hex
+let blob_digest = Blob.digest
+let string_of_blob = Blob.to_string
+
+(*****************************************************************************)
+(* Code *)
+(*****************************************************************************)
+
+(* A lot of this is from https://github.com/mirage/ocaml-git *)
+let load_store ?(path = Fpath.v (Sys.getcwd ())) () = Store.v path
+
+let load_store_exn ?(path = Fpath.v (Sys.getcwd ())) () =
+  match%lwt load_store ~path () with
+  | Ok store -> Lwt.return store
+  | Error (`Reference_not_found _) ->
+      failwith "Reference not found. Are you in a git repository?"
+  | Error (`Not_found _) -> failwith "Not found. Are you in a git repository?"
+  | Error (`Msg msg) -> failwith msg
+  | _ -> failwith "Unknown error"
```

Comment:
I think these should raise exceptions other than `Failure s` (*if* we really need an exception version, which I am skeptical of)

Reply:
I can change the error type if you lmk a better one. I like this pattern since there's nothing we can do historical wise if we can't find a store, so it makes sense to use this function imo